runhouse/resources/hardware/cluster_factory.py

```diff
@@ -95,260 +105,152 @@ def cluster(
         raise ValueError(
             "Cluster factory method can only accept one of `host` or `ips` as an argument."
         )
+    ips = kwargs.get("ips") or ([host] if isinstance(host, str) else host)
 
-    if name:
-        alt_options = dict(
-            host=host,
-            ssh_creds=ssh_creds,
-            server_port=server_port,
-            server_host=server_host,
-            server_connection_type=server_connection_type,
-            ssl_keyfile=ssl_keyfile,
-            ssl_certfile=ssl_certfile,
-            domain=domain,
-            den_auth=den_auth,
+    cluster_args = locals().copy()
```

Comment:
small nit and maybe just me, but was initially confused by the naming here, since `cluster_args` really behaves like kwargs

Reply:
yea using it as an easy way to get a dict out of the input args to easily pass around and use to check against cluster config, instead of manually creating a dict by passing in each arg as before (easy to miss an arg / forget to update when we add new factory args), but open to other approaches or naming, did you have one in mind?

Reply:
ya i like the way its extracted, its cleaner that way.. i think we can leave as is since it's not really "user facing" anyway